set_instance_tags/app.py

```diff
@@ -58,31 +62,93 @@ def get_principal_id(tags):
 
 def get_synapse_user_profile(synapse_id):
   '''Get synapse user profile data'''
-  syn = synapseclient.Synapse()
+  syn = get_synapse_client()
   user_profile = syn.getUserProfile(synapse_id)
   log.debug(f'Synapse user profile: {user_profile}')
-
   return user_profile
 
 
-def get_synapse_tags(user_profile):
-  '''Derive synapse tags from synapse user profile data'''
-
+def get_ssm_parameter(name):
```

Comment:
this code is repeated in set_bucket_tags/app.py and set_instance_tags/app.py.   Can you please factor it out so there is just one copy?

Reply:
good observation.  the AWS way to solve that is to setup [lambda layers](https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html) which is basically a lambda library that needs to be built and deployed separately.  I don't really like that approach and prefer to create a python library instead.  however i think a separate library is a bit overkill for this lambda at this time.  Therefore I've made an intentional decision to duplicate code instead of refactoring out to a lib.

Reply:
I don't (yet) understand.  It sounds like you are saying that you can't have more than one .py file for a Lambda, but we do this regularly in other Lambda's, e.g., Tess's  Budgets lambda: https://github.com/Sage-Bionetworks-IT/lambda-budgets/tree/master/budget. Why can't we do that here?  To make the suggestion more concrete, can't we have a file structure like:
```
set_tags/instance_app.py
set_tags/bucket_app.y
set_tags/common_python_code.py
```
and then in template.py use:
```
  SetBucketTagsFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: set__tags/
      Handler: bucket_app.handler
...
  SetInstanceTagsFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: set__tags/
      Handler: instance_app.handler
```
?

Reply:
thanks for the example.  My head was stuck on keeping the code in separate folders thus didn't see this solution.  I see your suggestion would be better.  This change would  be a large(ish) refactor.  Wondering if you prefer i refactor in this PR or should we merge and I do in a follow on PR?

Reply:
I don't want to slow things down but it would be easier/faster to review the code and tests with the duplicates removed.

Reply:
OK, it has been refactored to remove duplication.